go/vt/tabletserver/sqlquery.go

```diff
@@ -332,7 +335,15 @@ func handleExecError(query *proto.Query, err *error, logStats *SQLQueryStats) {
 			internalErrors.Add("Panic", 1)
 			return
 		}
-		*err = terr
+		if sq.config.TerseErrors {
+			if terr.SqlError == 0 {
```

Comment:
could combine the two ifs, if sq.config.TerseErrors && terr.SqlError != {...} else { *err = terr}